RoboJackets/robocup-software
# fix ubuntu-setup by moving git installs out

## Description
When installing our repo on a new machine, I realized why we had to git submodule update manually before after running ./util/ubuntu-setup. Turns out git stopped allowing their installs (including pre-commit) to be done by sudo. This PR fixes that by making a new util install script.

Updates:
 * changes default pip=22.1
 * requires new users to run ./util/git-setup after ubuntu-setup
 * fixes CI pipeline to account for this

## Steps to Test
 * delete local copy of this repo (MAKE SURE YOU DON'T HAVE ANY UNPUSHED CHANGES FIRST)
 * re-clone
 * checkout `fix-ubuntu-setup`
 * `./util/ubuntu-setup && ./util/git-setup`
 * `make clean && make perf`
 * `make run-sim`

Expected result: should work. (I have tested this on two machines, however, neither were fully clean when I tested. I suppose we'll find out when new members join in the fall.)
